pkg/helm/watcher/cache/cache.go

```diff
@@ -0,0 +1,182 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/gofrs/flock"
+
+	pb "github.com/weaveworks/weave-gitops/pkg/api/profiles"
+)
+
+const (
+	// we globally lock the whole base cache folder to avoid the
+	// possibility of checking if a sub-folder exists, just so it is created the moment
+	// we try to create it after we checked that it exists.
+	lockFilename    = "cache.lock"
+	profileFilename = "profiles.yaml"
+	valuesFilename  = "values.yaml"
+)
+
+// type alias for easier reading of the cache layout for values.
+type profileName = string
+type profileVersion = string
+
+// ValueMap contains easy access for a profile name and version based values file.
+type ValueMap map[profileName]map[profileVersion][]byte
+
+// Cache defines an interface to work with the profile data cacher.
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate . Cache
+type Cache interface {
+	Update(helmRepoNamespace, helmRepoName string, value Data) error
```

Comment:
bit nitty but isn;t this `CreateOrUpdate`? Or some better word

Reply:
Sure, why not. :) Technically, you update nothing to be something. :D

Reply:
Quite often Caches are `Put` (or `Set`) and `Get`

Reply:
Put is good too

Reply:
I like `Set`